src/platform/ESP32/ConfigurationManagerImpl.cpp

```diff
@@ -262,19 +263,27 @@ CHIP_ERROR ConfigurationManagerImpl::StoreCountryCode(const char * code, size_t
     return GenericConfigurationManagerImpl<ESP32Config>::StoreCountryCode(code, codeLen);
 }
 
+#if CHIP_DEVICE_CONFIG_ENABLE_ETHERNET
+CHIP_ERROR ConfigurationManagerImpl::GetPrimaryMACAddress(MutableByteSpan buf)
+{
+    return MapConfigError(esp_read_mac(buf.data(), ESP_MAC_ETH));
```

Comment:
needs a check that sufficient space available in buf.

Reply:
https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/misc_system_api.html#_CPPv412esp_read_macP7uint8_t14esp_mac_type_t

seems to say you need 6 or 8 bytes depending on `CONFIG_SOC_IEEE802154_SUPPORTED`

Reply:
actually you use ESP_MAC_ETH https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/misc_system_api.html#_CPPv4N14esp_mac_type_t11ESP_MAC_ETHE so you should check for buf length being at least 6 bytes (or maybe exactly 6 bytes or zero-fill the rest).

Reply:
added this change